specs/MultiProfile.md

```diff
@@ -0,0 +1,240 @@
+  /// CreateCoreWebView2ControllerWithOptions and CreateCoreWebView2CompositionControllerWithOptions.
+  HRESULT CreateCoreWebView2ControllerOptions(
+      [in] LPCWSTR profileName,
+      [in] BOOL inPrivateModeEnabled,
+      [out, retval] ICoreWebView2ControllerOptions** options);
+
+  /// Create a new WebView with options.
+  HRESULT CreateCoreWebView2ControllerWithOptions(
+      [in] HWND parentWindow,
+      [in] ICoreWebView2ControllerOptions* options,
+      [in] ICoreWebView2CreateCoreWebView2ControllerCompletedHandler* handler);
+
+  /// Create a new WebView in visual hosting mode with options.
+  HRESULT CreateCoreWebView2CompositionControllerWithOptions(
+      [in] HWND parentWindow,
+      [in] ICoreWebView2ControllerOptions* options,
+      [in] ICoreWebView2CreateCoreWebView2CompositionControllerCompletedHandler* handler);
+}
+
+[uuid(6E5CE5F0-16E6-4A05-97D8-4E256B3EB609), object, pointer_default(unique)]
+interface ICoreWebView2_7 : IUnknown {
+  /// The associated `ICoreWebView2Profile` object.
+  [propget] HRESULT Profile([out, retval] ICoreWebView2Profile** profile);
+}
+
+[uuid(3B9A2AF2-E703-4C81-9D25-FCE44312E960), object, pointer_default(unique)]
+interface ICoreWebView2Profile : IUnknown {
+  /// Name of the profile.
+  [propget] HRESULT ProfileName([out, retval] LPWSTR* value);
+
+  /// InPrivate mode is enabled or not.
+  [propget] HRESULT InPrivateModeEnabled([out, retval] BOOL* enabled);
+
+  /// Full path of the profile directory.
+  [propget] HRESULT ProfilePath([out, retval] LPWSTR* value);
+
+  // TODO: All profile-wide operations/settings will be put below in the future.
+}
+```
+
+## .NET and WinRT
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2ControllerOptions;
+    runtimeclass CoreWebView2Environment;
+    runtimeclass CoreWebView2;
+    runtimeclass CoreWebView2Profile;
+
+    runtimeclass CoreWebView2ControllerOptions
+    {        
+        String ProfileName { get; set; };
+
+        Int32 InPrivateModeEnabled { get; set; };
+    }
+    
+    runtimeclass CoreWebView2Environment
+    {
+        // ...
+    
+        CoreWebView2ControllerOptions CreateCoreWebView2ControllerOptions(
+            String ProfileName, Int32 InPrivateModeEnabled);
+        
+        Windows.Foundation.IAsyncOperation<CoreWebView2Controller>
+        CreateCoreWebView2ControllerWithOptionsAsync(
+            CoreWebView2ControllerWindowReference ParentWindow,
+            CoreWebView2ControllerOptions options);
+        
+        Windows.Foundation.IAsyncOperation<CoreWebView2CompositionController>
+        CreateCoreWebView2CompositionControllerWithOptionsAsync(
+            CoreWebView2ControllerWindowReference ParentWindow,
+            CoreWebView2ControllerOptions options);
+    }
+    
+    runtimeclass CoreWebView2
+    {
+        // ...
+        CoreWebView2Profile Profile { get; };
+    }
+    
+    runtimeclass CoreWebView2Profile
+    {
+        String ProfileName { get; };
```

Comment:
Do we have a way to know if a profile is the default profile?

Reply:
If this ProfileName is an empty string, it should be the Default profile unless "--profile-directory=xxx" command line switch is taking effect, in which case it would be the "xxx" profile specified there.

But checking the ProfilePath (the full path of the profile directory on disk) will help to know for sure whether it's the Default profile.

Reply:
Yeah I feel we might want to gradually remove the support for "--profile-directory=xxx" from now on, as this seems like a workaround in the previous era when we did not have multiple profile support. The only reason we're still keeping it is that we do not want to break some existing users who're potentially relying this workaround to not use the Default profile. Once it's gone, ProfileName being empty will indicate the Default profile for sure and the `IsDefault` will just be completely redundant.